src/config.rs

```diff
@@ -282,17 +280,44 @@ impl Config {
             ResolvAddr::resolve_or_exit,
         );
 
+        let changed_daemon_dir = config.daemon_dir != default_daemon_dir();
+
+        let mut default_cookie_path = config.daemon_dir;
+
         match config.network {
             Network::Bitcoin => (),
-            Network::Testnet => config.daemon_dir.push("testnet3"),
-            Network::Regtest => config.daemon_dir.push("regtest"),
-            Network::Signet => config.daemon_dir.push("signet"),
+            Network::Testnet => default_cookie_path.push("testnet3"),
+            Network::Regtest => default_cookie_path.push("regtest"),
+            Network::Signet => default_cookie_path.push("signet"),
             unsupported => unsupported_network(unsupported),
         }
+        default_cookie_path.push(".cookie");
+
+        if changed_daemon_dir {
+            eprintln!(
+                "Error: `daemon_dir` is not used anymore and deprecated, \
+                please remove this option."
+            );
+            if config.auth.is_none() && config.cookie_file.is_none() {
+                eprintln!(
+                    "Set the `cookie_file` option to \"{}\" because \
+                    `daemon_dir` will be removed in a future release.",
+                    default_cookie_path.display()
+                );
+            }
+            std::process::exit(1);
```

Comment:
I was going to suggest warning instead but since the messages are extra nice exit is better to ensure people see the nice ones. Also might make sense to exit after both messages were printed.

Reply:
> I was going to suggest warning instead but since the messages are extra nice exit is better to ensure people see the nice ones.

thought that as well

> Also might make sense to exit after both messages were printed.

done